core/ecschema-metadata/src/Metadata/Class.ts

```diff
@@ -640,6 +640,15 @@ export abstract class ECClass extends SchemaItem implements CustomAttributeConta
     return this.traverseBaseClassesSync(SchemaItem.equalByKey, targetClass);
   }
 
+  public static isECClass(object: any): object is ECClass {
+    if (!SchemaItem.isSchemaItem(object))
+      return false;
+
+    const objectType = (object as ECClass).schemaItemType;

```

Comment:
No need to as cast to ECClass here ... you know it's a schema item so you can get its schema item type without the as.